lib/routing_nb_config.c

```diff
@@ -0,0 +1,51 @@
+#include "northbound.h"
+#include "libfrr.h"
+#include "vrf.h"
+#include "routing_nb.h"
+
+
+/*
+ * XPath: /frr-routing:routing/control-plane-protocols/control-plane-protocol
+ */
+int routing_control_plane_protocols_control_plane_protocol_create(
+	enum nb_event event, const struct lyd_node *dnode,
+	union nb_resource *resource)
+{
+	struct vrf *vrf;
+	const char *vrfname;
+
+	switch (event) {
+	case NB_EV_VALIDATE:
+		vrfname = yang_dnode_get_string(dnode, "./vrf");
+		vrf = vrf_lookup_by_name(vrfname);
+		if (!vrf) {
+			zlog_warn("vrf is not configured\n");
```

Comment:
This is a pretty useless error message please clean it up to something useful

Reply:
done